autover/report.py

```diff
@@ -21,11 +18,24 @@ def autover(*packages):
                 ver = str(module.__version__)
             except Exception:
                 pass
-        
-        except (ImportError, ModuleNotFoundError):
+
+        except ImportError:
             try:
                 # See if there is a command by that name and check its --version if so
-                loc = subprocess.check_output(['which', package]).decode().strip()
+
+                # which not usually available on windows, though it
+                # could be (e.g. running in some kind of bash for
+                # windows, or which binary has been installed,
+                # etc...), so try it first. (And -a to match the list
+                # you get from where; might want to report more than
+                # one in the future if investigating confusion over
+                # what command runs when someone types 'command'.)
+                try:
+                    loc = subprocess.check_output(['which','-a',      package]).decode().splitlines()[0].strip()
+                except:
+                    # .exe in case powershell (otherwise wouldn't need it)
+                    loc = subprocess.check_output(['where.exe',       package]).decode().splitlines()[0].strip()
+
                 out = ""
                 try:
                     out = subprocess.check_output([package, '--version'], stderr=subprocess.STDOUT)
```

Comment:
Should we have a whitelist of commands here when actually running the command, basically anything that we ourselves need to query and that we might ask users for and that supports --version?  I'm thinking python, conda, jupyter, autover, and git, mainly; are there others?  It seems safer than just running an arbitrary command.